### PR TITLE
Fix alignment samples ordering

### PR DESCRIPTION
**Pull request by IpankajI**
This PR is also resolving #527

### CODE REVIEW

File: src/gui/mzroll/alignmentvizwidget.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ AlignmentVizWidget::AlignmentVizWidget(MainWindow* mw)
 void AlignmentVizWidget::plotGraph(PeakGroup*  group) {
 
     if (!_mw->alignmentVizDockWidget->isVisible()) return;
-
+    currentDisplayedGroup=group;
     intialSetup();
     PeakGroup grp = *group;
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/alignmentvizwidget.cpp]: @IpankajI Changing the variable names to reflect which group is the shadow group and which one is new would really help future developers

File: src/gui/mzroll/alignmentvizwidget.cpp [hunk 2]
```diff
@@ -212,7 +211,7 @@ void AlignmentVizWidget::plotIndividualGraph(PeakGroup group, QColor color) {
 
     int i = 1;
     Q_FOREACH(mzSample* sample, samples) {
-
+        QColor color = QColor(255*sample->color[0],255*sample->color[1],255*sample->color[2] ,alpha* sample->color[3]);
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/alignmentvizwidget.cpp]: Comments required here.

File: src/gui/mzroll/alignmentvizwidget.cpp [hunk 3]
```diff
@@ -254,7 +253,8 @@ vector<mzSample*> AlignmentVizWidget::getSamplesFromGroup(PeakGroup group) {
         mzSample* s = peaks[i].getSample();
         samples.push_back(s);
     }
-    sort (samples.begin(), samples.end());
+    sort (samples.begin(), samples.end(),mzSample::compSampleOrder);
+    reverse(samples.begin(),samples.end());
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/alignmentvizwidget.cpp]: Why reverse? Mention in comments maybe.

File: src/gui/mzroll/background_peaks_update.cpp [hunk 4]
```diff
@@ -485,14 +485,13 @@ void BackgroundPeakUpdate::align() {
...
                         QString errorMessage=QString::number(processedDataFromPython.size());
-                        errorMessage="Received data: "+errorMessage+" bytes";
 
                         if(!jDoc.isNull()){
                                 parentObj = jDoc.object();
                         }
                         else{
                                 if(processedDataFromPython.size()==0){
-                                        errorMessage=errorMessage+"<br>"+"Check parameters' value";
+                                        errorMessage=errorMessage + " Good groups found" +"<br>"+"Check parameters' value";
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/background_peaks_update.cpp]: Suggestion: "x good groups found. Relax parameters for better results"

File: src/gui/mzroll/background_peaks_update.cpp [hunk 5]
```diff
@@ -485,14 +485,13 @@ void BackgroundPeakUpdate::align() {
...
                                 }
                                 else{
                                         errorMessage=errorMessage+"<br>"+"Incomplete data";
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/background_peaks_update.cpp]: Error message should tell the user what to do when there is incomplete data. Should they run the algo again? Restart the program?

Comment (shubhra-agrawal) [reply src/gui/mzroll/background_peaks_update.cpp]: .

File: src/gui/mzroll/alignmentvizwidget.cpp (hunk 1 above)
Comment (IpankajI) [reply src/gui/mzroll/alignmentvizwidget.cpp]: @shubhra-agrawal changing.

File: src/gui/mzroll/alignmentvizwidget.cpp (hunk 2 above)
Comment (IpankajI) [reply src/gui/mzroll/alignmentvizwidget.cpp]: @shubhra-agrawal  adding

File: src/gui/mzroll/alignmentvizwidget.cpp (hunk 3 above)
Comment (IpankajI) [reply src/gui/mzroll/alignmentvizwidget.cpp]: It's making ordering of sample in Barplot and Ligand widget more consistent- both will have same color from upper row to lower row.

Comment (shubhra-agrawal) [reply src/gui/mzroll/alignmentvizwidget.cpp]: mention this in comments. This is not obvious.

File: src/gui/mzroll/background_peaks_update.cpp (hunk 4 above)
Comment (IpankajI) [reply src/gui/mzroll/background_peaks_update.cpp]: @shubhra-agrawal changed!

File: src/gui/mzroll/background_peaks_update.cpp (hunk 5 above)
Comment (IpankajI) [reply src/gui/mzroll/background_peaks_update.cpp]: @shubhra-agrawal updated!

**Pull request merged**